lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -71,16 +75,19 @@ def pass_hook_checks(self, event: EventBase) -> bool:
 
         return True
 
-    def _on_relation_joined(self, event) -> None:
+    def _on_relation_changed(self, event) -> None:
         """Handles providing mongos with KeyFile and hosts."""
         if not self.pass_hook_checks(event):
             logger.info("Skipping relation joined event: hook checks did not pass")
             return
 
         config_server_db = self.generate_config_server_db()
 
+        # create user and set secrets for mongos relation
+        self.charm.client_relations.oversee_users(None, None)
+
         # TODO Future PR, use secrets
-        self._update_relation_data(
+        self.update_relation_data(
             event.relation.id,
             {
                 KEYFILE_KEY: self.charm.get_secret(
```

Comment:
If you just create the `Requests` side with the `additional_secrets=[KEYFILE_KEY]` this would be automatically saved as a secret for you in the relation, once you set it via `update_relation_data()`.

Reply:
omg this is SO HELPFUL - let me do this as a separate PR on the `mongodb` side since the `mongodb` charm owns this lib